transaction/core/src/blockchain/block_version.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+use core::{fmt, hash::Hash, ops::Deref};
+use mc_crypto_digestible::Digestible;
+use serde::{Deserialize, Serialize};
+
+/// A block version number
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Digestible, Hash,
+)]
+pub struct BlockVersion(pub u32);
+
+impl From<u32> for BlockVersion {
+    fn from(src: u32) -> Self {
+        Self(src)
+    }
+}
+
+impl fmt::Display for BlockVersion {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl BlockVersion {
+    /// The maximum value of block_version that this build of
```

Comment:
```suggestion
    /// The minimum value of block_version that this build of
    /// mc-transaction-core has support for
    pub const MIN: Self = Self(1);

    /// The maximum value of block_version that this build of
```

Reply:
That is not the semantic of this field. If my transaction core has max block version 3, my software will handle any txout from versions 1 up to 3. Calling this a min implies that we won't be compatible with e.g. version 1 tx outs

Reply:
Oh I misunderstood

Reply:
I think this isn't actually the min, the origin block seems to have version 0

Reply:
OK, this was intended to replace `BlockVersion(1)`.

Reply:
Either way is fine by me.